File: Client/src/app/_models/message.ts
export interface message {
    id: string;
    senderId: string;
    senderUsername: string;
    senderPhotoUrl: string;
    recipientId: string;
    recipientUsername: string;
    recipientPhotoUrl: string;
    content: string;
    messageSent: Date;
    dateRead?: any;
}

File: API/Entity/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entity
{
    public class Message
    {
        [Key]
        public Guid Id {get; set;}
        public Guid SenderId {get; set;}
        public string SenderUsername {get; set;}
        public AppUser Sender {get; set;}

        public Guid RecipientId {get; set;}
        public string RecipientUsername {set; get;}
        public AppUser Recipient {get; set;}
        

        public string Content {get; set;}


        public DateTime? MessageSent {get; set;} = DateTime.Now;
        public DateTime? DateRead {get; set;}
        public bool? SenderDeleted {get; set;}
        public bool? RecipientDeleted {get; set;}

    }
}